service/controller/v18/resource/ipam/create.go

```diff
@@ -104,13 +128,29 @@ func (r *Resource) allocateSubnet(ctx context.Context) (string, error) {
 
 		subnet, err = ipam.Free(r.networkRange, r.allocatedSubnetMask, reservedSubnets)
 		if err != nil {
-			return "", microerror.Maskf(err, "networkRange: %s, allocatedSubnetMask: %s, reservedSubnets: %#v", r.networkRange.String(), r.allocatedSubnetMask.String(), reservedSubnets)
+			return net.IPNet{}, microerror.Maskf(err, "networkRange: %s, allocatedSubnetMask: %s, reservedSubnets: %#v", r.networkRange.String(), r.allocatedSubnetMask.String(), reservedSubnets)
 		}
 
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found free subnet %#q", subnet.String()))
 	}
 
-	return subnet.String(), nil
+	return subnet, nil
+}
+
+func (r *Resource) selectRandomAZs(n int) ([]string, error) {
```

Comment:
IMO this deserves a unit test. When we make the seeding configurable this is easily doable and ensures the randomization works properly when seeded accordingly.

Reply:
I think unit test for this can be implemented without dragging unnecessary type along. We don't need to care about which specific zones get selected, just that they are different on each call. I'll write this.

Reply:
See the test.